pyspeckit/spectrum/readers/gbt.py

```diff
@@ -98,9 +117,17 @@ def read_gbt_scan(sdfitsfile, obsnumber=0):
     # Convert xarr to LSR units
     #sp.xarr.convert_to_unit('m/s')
     obsfreq = header['OBSFREQ']
-    centerfreq = pyspeckit.spectrum.units.SpectroscopicAxis(np.float(obsfreq),'Hz',refX=obsfreq,refX_unit='Hz')
-    delta_freq = (centerfreq.as_unit('m/s') + header['VFRAME']).as_unit(centerfreq.units) - centerfreq
-    sp.xarr -= delta_freq
+    
+    # some creative abuse how a S.A. stores and converts units, just in order to convert VFRAME in m/s to a delta_freq in Hz
+    c1 = pyspeckit.spectrum.units.SpectroscopicAxis(obsfreq,'Hz',velocity_convention='radio',
+                                                    refX=obsfreq, refX_unit=sp.xarr.unit)
+    v1 = (c1.as_unit('m/s') + u.Quantity(header['VFRAME'],'m/s')).value
```

Comment:
I think this section should just be simplified.  This is a lot of complexity that is not justified.
```
centerfreq = u.Quantity(obsfreq, u.Hz)
delta_freq = u.Quantity(header['VFRAME'], u.m/u.s).to(u.Hz, u.doppler_radio(centerfreq)) - centerfreq
sp.xarr = sp.xarr - delta_freq
```